saphir/src/redirect.rs

```diff
@@ -74,9 +83,20 @@ impl Builder {
     }
 
     #[inline]
-    pub fn add_header<K: AsRef<str>, V: AsRef<str>>(mut self, key: K, value: V) -> Self {
-        self.extra_headers.insert(key.as_ref().to_owned(), value.as_ref().to_owned());
-        self
+    pub fn header<K, V>(mut self, key: K, value: V) -> Result<Self, BuilderError>
```

Comment:
Error are returned when calling the `build` fn like on the response::Builder type